### PR TITLE
HumanBodymap Component #84

### LINKED ISSUE #123: 💡[FEATURE]:  Integration Of Components: Create ArticleDescriptionScreen
## Task Area:

./frontend/src/screens/

## Task Description

Create ArticleDescription Screen.

Complete the rest design.
Implement Text Watcher functionality

image

Give Validation for each field.

Add it to StackNavigation, and Design the screen  as per UI and with above components. (NoteIcon Click Action)

image

### PR DESCRIPTION
**Pull request by Ar7109**
#### PR Description
Issue #120 

- Fixed a module resolution error in App.tsx.
- Updated the HumanBodyMap component to replace deprecated defaultProps usage with default parameters.
- Added proper import paths to ensure components are correctly linked.
- Cleared Metro Bundler cache to resolve potential caching issues.
- Improved component structure and readability in HumanBodyMap.

#### Type of Change
- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Documentation update

#### Select your work-area
- [x] Frontend
- [ ] Backend
- [ ] Documentation
- [ ] Others

#### Related Issue
This PR resolves issue #123.

#### Add your Work Example
image

#### Fixes (mention the issue number which this fixes)
#closes #123

#### Checklist
- [x] I have updated my branch and synced it with the project's 'develop' branch before making this PR.
- [x] I have optimized the file changes.
- [x] I have added a snapshot of my work example.
- [x] I have made a PR to the project's develop branch.

#### Undertaking
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] I have checked for plagiarism and assure its authenticity.
- [x] I have read and followed the code of conduct for this repository. I understand that violation of this undertaking may have legal consequences.

- [x] I Agree

### CODE REVIEW

File: frontend/src/components/HumanBodyMap.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import React, { useState } from "react";
+import { StyleSheet, Switch, Text, View } from "react-native";
+import Body from "react-native-body-highlighter";
+
+const HumanBodyMap = ({ 
+  onBodyPartSelect = () => {} // Use default parameter
```
Comment (SB2318) [inline frontend/src/components/HumanBodyMap.js:6]: Complete the function, Add a TextView (With proper style) above the bodymap, your selection will be reflect there.

File: frontend/src/components/HumanBodyMap.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+}) => {
+  const [isBackSideEnabled, setIsBackSideEnabled] = useState(false);
+  const [isMale, setIsMale] = useState(true);
```
Comment (SB2318) [inline frontend/src/components/HumanBodyMap.js:9]: No need

File: frontend/src/components/HumanBodyMap.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import React, { useState } from "react";
+import { StyleSheet, Switch, Text, View } from "react-native";
+import Body from "react-native-body-highlighter";
+
+const HumanBodyMap = ({ 
+  onBodyPartSelect = () => {} // Use default parameter
+}) => {
+  const [isBackSideEnabled, setIsBackSideEnabled] = useState(false);
```
Comment (SB2318) [inline frontend/src/components/HumanBodyMap.js:8]: Only keep this part

File: frontend/src/components/HumanBodyMap.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  const toggleGenderSwitch = () => setIsMale((previousState) => !previousState);
+
+  const handleBodyPartPress = (e) => {
+    onBodyPartSelect(e);
+  };
+
+  return (
+    <View style={styles.container}>
+      <Body
+        data={[
```
Comment (SB2318) [inline frontend/src/components/HumanBodyMap.js:21]: Add all the available body part

File: frontend/src/components/HumanBodyMap.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+          { slug: "chest", intensity: 1 },
+          { slug: "abs", intensity: 2 },
+          { slug: "upper-back", intensity: 1 },
+          { slug: "lower-back", intensity: 2 },
+          { slug: "biceps", intensity: 2 }, // Add more body parts if needed
+        ]}
+        onBodyPartPress={handleBodyPartPress}
+        gender={isMale ? "male" : "female"}
```
Comment (SB2318) [inline frontend/src/components/HumanBodyMap.js:29]: Cancel it

**Review by SB2318 — COMMENTED** [review]
Update your **develop** branch first, 
Checkout develop branch
Add your changes with requested changes
and then Create PR against develop branch.

image

**Comment by Ar7109** [comment]
I have done the changes, once review it and suggest for correction and further improvement

**Comment by SB2318** [comment]
PR against SB2318develop branch.
image

@Ar7109

**Comment by SB2318** [comment]
@Ar7109 please create PR for  **develop** branch.

**Comment by SB2318** [comment]
@Ar7109, please let me know if you're interested in merging your solution. If you need any guidance from my side to create a successful pull request (PR), cause **GSSOC** program ending soon.

**Pull request closed without merging**